modules/adipoloBidAdapter.js

```diff
@@ -1,37 +1,26 @@
 import {BANNER, VIDEO} from '../src/mediaTypes.js';
 import {registerBidder} from '../src/adapters/bidderFactory.js';
-import {buildRequests, getUserSyncs, interpretResponse} from '../libraries/xeUtils/bidderUtils.js';
-import {deepAccess, getBidIdParameter, isArray, logError} from '../src/utils.js';
+import {buildRequests, getUserSyncs, interpretResponse, isBidRequestValid} from '../libraries/xeUtils/bidderUtils.js';
 
 const BIDDER_CODE = 'adipolo';
-const ENDPOINT = 'https://prebid.adipolo.live';
-const GVL_ID = 1456;
-
-function isBidRequestValid(bid) {
-  if (bid && typeof bid.params !== 'object') {
-    logError('Params is not defined or is incorrect in the bidder settings');
-    return false;
-  }
-
-  if (!getBidIdParameter('pid', bid.params)) {
-    logError('Pid is not present in bidder params');
-    return false;
-  }
 
-  if (deepAccess(bid, 'mediaTypes.video') && !isArray(deepAccess(bid, 'mediaTypes.video.playerSize'))) {
-    logError('mediaTypes.video.playerSize is required for video');
-    return false;
-  }
+const ENDPOINTS = {
+  'us-east': 'https://prebid.adipolo.live',
+  'eu-center': 'https://prebid-eu.adipolo.live',
+};
 
-  return true;
-}
+const GVL_ID = 1456;
 
 export const spec = {
   code: BIDDER_CODE,
   gvlid: GVL_ID,
   supportedMediaTypes: [BANNER, VIDEO],
-  isBidRequestValid,
-  buildRequests: (validBidRequests, bidderRequest) => buildRequests(validBidRequests, bidderRequest, ENDPOINT),
+  isBidRequestValid: bid => isBidRequestValid(bid, ['pid']),
+  buildRequests: (validBidRequests, bidderRequest) => {
+    const region = validBidRequests[0]?.params.ext?.region;
```

Comment:
why not just detect the eu like this https://github.com/prebid/Prebid.js/blob/fb966a2da45da3e42d54a25f9f9a17b5d0374cfb/modules/escalaxBidAdapter.js#L55 ; or at least fallback to that instead of the us